controllers/listingskills.js

```diff
@@ -0,0 +1,128 @@
+const { db } = require('../config/db');
+const asyncHandler = require('../middleware/async');
+const ErrorResponse = require('../utils/errorResponse');
+
+/**
+ * @desc    Get all listing skills
+ * @route   GET /api/listing-skills
+ * @desc    Get all listing skills for a listing
+ * @route   GET /api/listings/:listing_id/listing-skills
+ * @access  Public
+ */
+exports.getListingSkills = asyncHandler(async (req, res, next) => {
+  if (req.params.listing_id) {
+    // return 404 error response if listing not found
+    const skills = await db.many(
+      'SELECT ls.listing_id, ls.listing_skill_id, ls.skill_id, s.skill FROM ListingSkills ls LEFT JOIN Skills s ON ls.skill_id = s.skill_id WHERE ls.listing_id = $1',
```

Comment:
From @Kalsyc : consider
`SELECT ls.listing_skill_id, ls.listing_id, ls.skill_id, s.skill FROM ListingSkills ls INNER JOIN Skills s ON ls.skill_id = s.skill_id WHERE ls.listing_id = $1`

to achieve 3 outcomes:
1. listing id does not exist -> 404
2. listing id exist, but 0 listing skills  -> 200, count:0
3. listing id exist and x listing skills -> 200, count:x